### PR TITLE
#167 wrtc fix and updated examples to work in esm

### PR DESCRIPTION
**Pull request by konsumer**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[X] Bug fix
[ ] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**

I swapped out `wrtc` with `@koush/wrtc` which works on Mac M1s, and builds (currently there is a node-gyp error in wrtc, #167 ) Additionally, the examples use `require` and `__filename` even though the lib is set to `type: module` so they wouldn't run (and even renaming them to .cjs didn't work, so I just updated them to use esm syntax.)

**Which issue (if any) does this pull request address?**

#167

### CODE REVIEW

**Comment by ThaUnknown** [comment]
did you test if this can connect with web peers?

File: examples/node-seed.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
-const WebTorrent = require('../index')
+import WebTorrent from '../index.js'
+import { fileURLToPath } from 'url'
+import { dirname } from 'path'
```
Comment (ThaUnknown) [inline examples/node-seed.js:3]: unused

**Comment by konsumer** [comment]
> did you test if this can connect with web peers?

yep, it connects with web-peers

File: examples/node-seed.js (hunk 1 above)
Comment (konsumer) [reply examples/node-seed.js]: Ah, yeh, don't need `__dirname`. removed.

**Comment by ThaUnknown** [comment]
yeah, this seems like a potential improvement, and might be a good idea, but there's a lot of shit to be looked into, so it's likely this will go stale for quite a while

important shit that needs to be done original vs this PR:

NWjs support comparisons
Electron support comparisons
performance with web peers
speeds with web peers

**Comment by konsumer** [comment]
@ThaUnknown  Yeh, I hear that. At first I was using my fork of this, but in my project I just follow the above, since I'd rather not have a dep on some random fork (even if it's mine) and it means I can use latest published webtorrent, instead of locking into whatever this has deps on. This PR solves the original problem, for others, of this library not being installable.

**Comment by ThaUnknown** [comment]
you can run these tests yourself and posts results here is you actually want this merged

**Comment by konsumer** [comment]
> you can run these tests yourself and posts results here is you actually want this merged

I am not sure I understand. What are "these tests"? I only updated the broken dep, and fixed the examples. I am using webtorrent without this wrapper, so if you don't want it, don't merge it.

**Comment by ThaUnknown** [comment]
> > you can run these tests yourself and posts results here is you actually want this merged
> 
> I am not sure I understand. What are "these tests"? I only updated the broken dep, and fixed the examples. I am using webtorrent in node (working with web client, etc) without this wrapper, so if you don't want it, don't merge it.

> NWjs support comparisons
> Electron support comparisons
> performance with web peers
> speeds with web peers

comparisons before and after these changes, to check if these changes don't remove support from places where webtorrent might have been used, so stuff like electron  >=16.0, node arm etc

**Comment by konsumer** [comment]
> > NWjs support comparisons
> > Electron support comparisons
> > performance with web peers
> > speeds with web peers
> 
> comparisons before and after these changes, to check if these changes don't remove support from places where webtorrent might have been used, so stuff like electron >=16.0, node arm etc

I don't see how this would effect NWjs, electron, etc (they use webtorrent direct with WebRTC in browser, not this wrapper, right?)

webtorrent-hybrid does not install at all on my M1, or any modern version of node. Do you have tests for these things for web speed-comparisons with the old webrtc lib? If you have a look at [@koush/wrtc](https://github.com/koush/node-webrtc), it's a pretty minimal fork that just updated a few things to make it build.

Overall, I'm not really sure why this is something that you expect me to test for a little wrapper thing I am not using. I don't even use electron or nwjs (it's been years.)

I thought it might help others to make webtorrent-hybrid work, since it's still in various instructions for "do webtorrent in node". That's me just trying to help out, but again, if you're not interested in webtorrent-hybrid building on M1s or any modern version of node (I think < v10 might have issues), then that's totally cool with me. I don't really understand, but you're the one with commit-rights, and I can happily use webtorrent without it.

As you can see with the linked issue above, I also am suggesting it be added to webtorrent-cli, since it also cannot seed for web-peers, currently, and it would be handy to be able to use the official CLI to do this.

**Comment by konsumer** [comment]
@feross this is similar to my other PR. It's basically just an update to the WebRTC polyfill to make it work again.

**Comment by haywirez** [comment]
@konsumer I was facing the same issues (building on M1, trying to use current node v19) for a new project. Can confirm your fork seems to work with web peers but proper testing of `@koush/node-webrtc`  is near impossible on this platform as neither the original `node-webrtc` (which hasn't been updated since 2021) or the fork repo builds on an M1. Someone with a different setup or VMs could try.

I don't know how we'd do E2E tests within this project space as `webtorrent-desktop` uses the `webtorrent` package, not `webtorrent-hybrid`...

**Pull request closed without merging**